stackslib/src/config/mod.rs

```diff
@@ -1210,53 +1210,359 @@ impl std::default::Default for Config {
+    /// default configurations, genesis block definitions, and overall node behavior.
+    ///
+    /// Supported values:
+    /// ```text
+    /// | Mode            | Network Type |
+    /// |-----------------|--------------|
+    /// | "mainnet"       | mainnet      |
+    /// | "xenon"         | testnet      |
+    /// | "mocknet"       | regtest      |
+    /// | "helium"        | regtest      |
+    /// | "neon"          | regtest      |
+    /// | "argon"         | regtest      |
+    /// | "krypton"       | regtest      |
+    /// | "nakamoto-neon" | regtest      |
+    /// ```
+    ///
+    /// Default: `"mocknet"`
     pub mode: String,
+    /// The network-specific identifier used in P2P communication and database initialization.
+    /// Derived from `mode` during config load, unless explicitly overridden (for test purposes).
+    ///
+    /// **Warning:** Do not modify this unless you really know what you're doing.
+    /// This is intended strictly for testing purposes.
+    ///
+    /// Default: Derived from [`BurnchainConfig::mode`] ([`CHAIN_ID_MAINNET`] for `mainnet`, [`CHAIN_ID_TESTNET`] otherwise).
     pub chain_id: u32,
+    /// The peer protocol version number used in P2P communication.
+    /// This parameter cannot be set via the configuration file.
+    ///
+    /// **Warning:** Do not modify this unless you really know what you're doing.
+    ///
+    /// Default: Derived from [`BurnchainConfig::mode`] ([`PEER_VERSION_MAINNET`] for `mainnet`, [`PEER_VERSION_TESTNET`] otherwise).
     pub peer_version: u32,
+    /// Specifies a mandatory wait period (in milliseconds) after receiving a burnchain tip
+    /// before the node attempts to build the anchored block for the new tenure.
+    /// This duration effectively schedules the start of the block-building
+    /// process relative to the tip's arrival time.
+    ///
+    /// This is intended strictly for testing purposes.
+    ///
+    /// Default: `5000` milliseconds.
+    ///          `10000` milliseconds when launched with the `helium` or `mocknet` subcommands.
     pub commit_anchor_block_within: u64,
+    /// The Maximum amount (in sats) of "burn commitment" to broadcast for the next block's leader election.
+    /// Acts as a safety cap to limit the maximum amount spent on mining.
+    /// It serves as both the target fee and a fallback if dynamic fee calculations fail or cannot be performed
+    ///
+    /// This setting can be hot-reloaded from the config file, allowing adjustment without restarting.
+    ///
+    /// Only relevant if [`NodeConfig::miner`] is `true`.
+    ///
+    /// Default: `20000` satoshis
     pub burn_fee_cap: u64,
+    /// The hostname or IP address of the bitcoin node peer.
+    ///
+    /// This field is required for all node configurations as it specifies where to find the underlying
+    /// bitcoin node to interact with for PoX operations, block validation, and mining.
+    ///
+    /// Default: `"0.0.0.0"`
```

Comment:
hmm, if the default is really 0.0.0.0 - can you open an issue to change that to 127.0.0.1? we definitely not default to 0's here.